libs/types/src/serviceTypes.ts

```diff
@@ -0,0 +1,44 @@
+interface SuccessStatus<T> {
+  status: 'SUCCESS';
+  data: T;
+}
+interface ErrorStatus<E> {
+  status: 'ERROR';
+  type: E;
+}
+
+export type Status<T, E> = SuccessStatus<T> | ErrorStatus<E>;
+export type AsyncStatus<T, E> = Promise<Status<T, E>>;
+
+export const success = <T>(data: T): SuccessStatus<T> => ({
+  status: 'SUCCESS',
+  data,
+});
+
+export const error = <E>(error: E): ErrorStatus<E> => ({
+  status: 'ERROR',
+  type: error,
+});
+
+export const isSuccess = <T, E>(
+  status: Status<T, E>
+): status is SuccessStatus<T> => status.status === 'SUCCESS';
+
+export const successMap = <T1, E, T2>(
+  status: Status<T1, E>,
+  f: (s: T1) => T2
+): Status<T2, E> =>
+  status.status === 'SUCCESS' ? success(f(status.data)) : status;
+
+export const isError = <T, E>(status: Status<T, E>): status is ErrorStatus<E> =>
+  status.status === 'ERROR';
+
+export const mapStatusSuccess = <T, E, U>(
+  data: Status<T, E>,
+  f: (data: T) => U
+): Status<U, E> => {
+  if (data.status === 'SUCCESS') {
+    return { status: 'SUCCESS', data: f(data.data) };
+  }
+  return data;
+};
```

Comment:
I'm not sure what to do with these – they are strongly connected to the Status types, but they should probably not live inside the `@dundring/types` package (?)

Any thoughts would be greatly appreciated:))

Reply:
Maybe the "types" and "utils" pacakage should be merged? Might not be necessary to have a separate pacakage for types 🤔